rotkehlchen/externalapis/alchemy.py

```diff
@@ -0,0 +1,400 @@
+        else:
+            price = usd_price
+
+        GlobalDBHandler.add_historical_prices(entries=[HistoricalPrice(
+            from_asset=from_asset,
+            to_asset=to_asset,
+            source=HistoricalPriceOracle.ALCHEMY,
+            timestamp=timestamp,
+            price=price,
+        )])
+        return price
+
+    def can_query_history(
+            self,
+            from_asset: Asset,  # pylint: disable=unused-argument
+            to_asset: Asset,
+            timestamp: Timestamp,
+            seconds: int | None = None,
+    ) -> bool:
+        return not self.is_penalized()
+
+    @staticmethod
+    def _get_asset_identifier(asset: AssetWithOracles) -> dict[str, str]:
+        """
+        May raise:
+        - UnsupportedAsset
+        """
+        if asset.is_evm_token() is True:
+            asset = asset.resolve_to_evm_token()
+            if asset.chain_id == ChainID.POLYGON_POS:
+                network = 'polygon-mainnet'
+            elif asset.chain_id == ChainID.ARBITRUM_ONE:
+                network = 'arb-mainnet'
+            elif asset.chain_id == ChainID.BINANCE:
+                network = 'bnb-mainnet'
+            elif asset.chain_id == ChainID.AVALANCHE:
+                network = 'avax-mainnet'
+            elif asset.chain_id == ChainID.ZKSYNC_ERA:
+                network = 'zksync-mainnet'
+            elif asset.chain_id == ChainID.ETHEREUM:
+                network = 'eth-mainnet'
+            elif asset.chain_id == ChainID.POLYGON_ZKEVM:
+                network = 'polygonzkevm-mainnet'
+            elif asset.chain_id == ChainID.SCROLL:
+                network = 'scroll-mainnet'
+            elif asset.chain_id == ChainID.GNOSIS:
+                network = 'gnosis-mainnet'
+            elif asset.chain_id == ChainID.BASE:
+                network = 'base-mainnet'
+            elif asset.chain_id == ChainID.FANTOM:
+                network = 'fantom-mainnet'
+            elif asset.chain_id == ChainID.ARBITRUM_NOVA:
+                network = 'arbnova-mainnet'
+            else:
+                raise UnsupportedAsset(asset.identifier)
+
+            return {'network': network, 'address': asset.evm_address}
+
+        # Alchemy team recommends using standard ticker symbols (e.g. ETH)
+        # over coingecko ids (e.g. ethereum) for better reliability.
+        return {'symbol': asset.symbol}
+
+    def _deserialize_current_price(
+            self,
+            result: list[dict[str, Any]],
+            asset_identifier: dict[str, str],
+            from_asset: Asset,
+            to_asset: Asset,
+    ) -> Price:
+        """Deserializes current price data from Alchemy API response.
+
+        If price data is missing, invalid, or there's an error, ZERO_PRICE is returned.
+        Otherwise, the deserialized price is returned.
+        """
+        try:
+            for entry in result:
+                if self._is_identifier_match(entry, asset_identifier) is False:
```

Comment:
For boolean stuff keep it simple. I won't repeat further down, but please do it in all other places

```suggestion
                if not self._is_identifier_match(entry, asset_identifier):
```